packages/build-tools/src/android/gradle.ts

```diff
@@ -35,4 +36,54 @@ export async function runGradleCommand(
     },
     env: ctx.env,
   });
+  if (ctx.isCloudBuild && process.platform === 'linux') {
+    adjustOOMScore(spawnPromise, ctx.logger);
+  }
+
+  await spawnPromise;
+}
+
+/**
+ * OOM Killer sometimes kills worker server while build is exceeding memory limits.
+ * `oom_score_adj` is a value between -1000 and 1000 that defines which process
+ * is more likely to get killed(higher value more likely).
+ *
+ * This function sets oom_score_adj for gradle process and all it's child processes.
+ */
+function adjustOOMScore(spawnPromise: SpawnPromise<SpawnResult>, logger: bunyan): void {
+  setTimeout(
+    async () => {
+      logger.info(spawnPromise.child.pid);
+      try {
+        const children: number[] = [spawnPromise.child.pid!];
```

Comment:
Maybe use `assert` instead of `!`?

Reply:
good catch, I added that and log above for testing and forgot to change it